src/api.hpp

```diff
@@ -1,4 +1,6 @@
 #include "task_parameters.hpp"
+#include "tree.hpp"
```

Comment:
These bring in C++ headers, which break the C-API test. At the moment, I'm unsure how to correctly address this. I don't believe we can just wrap the `include`s in `#if def __cplusplus` in `api.hpp` as that would leave `biom` and `BPTree` undefined, and we cannot do it within `biom.hpp` and `tree.hpp` as that would leave components of their signatures undefined.

Reply:
...the present solution is to limit use of `one_off_inmem` to c++...